easybuild/tools/modules.py

```diff
@@ -156,8 +156,8 @@ def __init__(self, mod_paths=None, testing=False):
 
         # actual module command (i.e., not the 'module' wrapper function, but the binary)
         self.cmd = self.COMMAND
-        if self.COMMAND_ENVIRONMENT is not None and self.COMMAND_ENVIRONMENT in os.environ:
-            self.log.debug('Set command via environment variable %s' % self.COMMAND_ENVIRONMENT)
+        if which(self.cmd) is None and self.COMMAND_ENVIRONMENT in os.environ:
```

Comment:
i would log the case where `which(self.cmd)` and `self.COMMAND_ENVIRONMENT` are not equal (debug is enough, but can help debugging with messed up setups)